base/applications/cmdutils/taskkill/taskkill.c

```diff
@@ -440,6 +605,167 @@ static BOOL add_to_task_list(WCHAR *name)
+                }
+                force_termination = TRUE;
+                break;
+            }
+            case OP_PARAM_IMAGE:
+            case OP_PARAM_PID:
+            {
+                if (!argv[i + 1])
+                {
+                    taskkill_message_printfW(STRING_MISSING_PARAM, argv[i]);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (argument == OP_PARAM_IMAGE)
+                    has_im = TRUE;
+                if (argument == OP_PARAM_PID)
+                    has_pid = TRUE;
+
+                if (has_im && has_pid)
+                {
+                    taskkill_message(STRING_MUTUAL_EXCLUSIVE);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (get_argument_type(argv[i + 1]) != OP_PARAM_INVALID)
+                {
+                    taskkill_message_printfW(STRING_MISSING_PARAM, argv[i]);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+
+                if (!add_to_task_list(argv[++i])) // add next parameters to task_list
+                    return FALSE;
+
+                break;
+            }
+            case OP_PARAM_HELP:
+            {
+                if (has_help == TRUE)
+                {
+                    // -? already specified
+                    taskkill_message_printfW(STRING_PARAM_TOO_MUCH, argv[i], 1);
+                    taskkill_message(STRING_USAGE);
+                    return FALSE;
+                }
+                has_help = TRUE;
+                break;
+            }
+            case OP_PARAM_TERMINATE_CHILD:
+            {
+                WINE_FIXME("argument T not supported\n");
+                break;
+            }
+            case OP_PARAM_INVALID:
+            default:
+            {
+                taskkill_message(STRING_INVALID_OPTION);
+                taskkill_message(STRING_USAGE);
+                return FALSE;
+            }
+            }
+        }
+    }
+
+    if (has_help)
+    {
+        if (argc > 2) // any parameters other than -? is specified
+        {
+            taskkill_message(STRING_INVALID_SYNTAX);
+            taskkill_message(STRING_USAGE);
+            return FALSE;
+        }
+        else
+        {
+            taskkill_message(STRING_USAGE);
+            exit(0);
+        }
+    }
+    else if ((!has_im) && (!has_pid)) // has_help == FALSE
+    {
+        // both has_im and has_pid are missing (maybe -fi option is missing too, if implemented later)
+        taskkill_message(STRING_MISSING_OPTION);
+        taskkill_message(STRING_USAGE);
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+int wmain(int argc, WCHAR* argv[])
+{
+    int status_code = 0;
+
+    if (!process_arguments(argc, argv))
+    {
+        HeapFree(GetProcessHeap(), 0, task_list);
+        return 1;
+    }
+
+    /*  if (force_termination)
```

Comment:
You'll want to do the ifdefs more granularly if you can. For example, you can reduce it down to one `wmain` function and simply use 
```
#ifndef __REACTOS__
    if (force_termination)
        status_code = terminate_processes();
    else
        status_code = SendCloseMessages();
#endif
```